core/qbft/qbft.go

```diff
@@ -27,34 +26,42 @@ import (
 	"github.com/obolnetwork/charon/app/errors"
 )
 
+// Value defines the generic value type constraint
+// that provides an equality method.
+type Value[V any] interface {
+	Equal(V) bool
+}
+
 // Transport abstracts the transport layer between processes in the consensus system.
 //
 // Note that broadcasting doesn't return an error. Since this algorithm is idempotent
 // it is suggested to just retry broadcasting indefinitely until it succeeds or times out.
-type Transport struct {
+type Transport[I any, V Value[V]] struct {
```

Comment:
@corverroos what is I and what is V ?

Reply:
`Instance` and `Value` https://github.com/ObolNetwork/charon/pull/501/files#diff-675cdeeca6e8a606721a8f590fe58fbae35d8ba37a64956b018f047ace94b541R62

Reply:
Yup:
```
// Instance identifies the consensus instance.
Instance() I

// Value being proposed.
Value() V
```